File: resources/pi_calculator.py
#!/usr/bin/env python3
# This is a pi calculation script
import time
import argparse

def calculate(iterations, p1, p2):
  result = 1.0
  for i in range(1, iterations + 1):
    j = i *  p1 - p2
    result -= (1/j)
    j = i * p1 + p2
    result += (1/j)
  return result

def main():
  parser = argparse.ArgumentParser(description="Calculate a mathematical series.")
  parser.add_argument('--iterations', type=int, default=100000000, help='Number of iterations to perform')
  args = parser.parse_args()

  # Use the provided iterations value
  iterations = args.iterations
  start_time = time.time()
  result = calculate(iterations, 4, 1) * 4
  end_time = time.time()
  print(f"Result: {result:.12f}")
  print(f"Execution time: {end_time - start_time:.6f} seconds")

if __name__ == "__main__":
  main()

File: resources/pi_calculator.cpp


#include <iostream>
#include <chrono>
#include <cmath>

// Function to calculate the series using iterations, p1, and p2
double calculate(int iterations, int p1, int p2) {
    double result = 1.0;
    for (int i = 1; i <= iterations; ++i) {
        double j = i * p1 - p2;
        result -= (1.0 / j);
        j = i * p1 + p2;
        result += (1.0 / j);
    }
    return result;
}

int main(int argc, char *argv[]) {
    int iterations = 100000000; // Default number of iterations
    if (argc > 1) {
        iterations = std::stoi(argv[1]); // Read iterations from command line arguments if provided
    }

    auto start_time = std::chrono::high_resolution_clock::now(); // Start timing
    double result = calculate(iterations, 4, 1) * 4; // Call calculation function
    auto end_time = std::chrono::high_resolution_clock::now(); // End timing
    std::chrono::duration<double> elapsed = end_time - start_time; // Calculate elapsed time

    std::cout.precision(12);
    std::cout << "Result: " << result << std::endl;
    std::cout << "Execution time: " << elapsed.count() << " seconds" << std::endl; // Output execution time

    return 0;
}
